docs/schema-changes-use-cases.md

```diff
@@ -0,0 +1,605 @@
+
+```yaml
+# Version n
+groups:
+  - id: registry.process
+    type: attribute_group
+    brief: "Process attributes."
+    attributes:
+      - id: process.cpu.state
+        brief: "The state of the CPU"
+        type: ...
+        stability: experimental
+```
+
+To rename the `process.cpu.state` attribute to `cpu.mode`, the following changes are made:
+
+```yaml
+# Version n+1
+groups:
+  - id: registry.process.deprecated
+    type: attribute_group
+    brief: Attributes specific to a cpu instance.
+    attributes:
+      # The existing attribute is deprecated
+      - id: process.cpu.state
+        brief: "Deprecated, use `cpu.mode` instead."
+        deprecated:
+          type: renamed
+          new_name: cpu.mode
+        type: ...
+        stability: experimental
+  
+  - id: registry.cpu
+    type: attribute_group
+    brief: Attributes specific to a cpu instance.
+    display_name: CPU Attributes
+    attributes:
+      # A new attribute is introduced
+      - id: cpu.mode
+        brief: "The mode of the CPU"
+        type: ...
+        stability: experimental
+        examples: [ "user", "system" ]
+``` 
+
+From these changes, Weaver can generate a diff report that shows the renaming of the
+`process.cpu.state` attribute to `cpu.mode`.
+
+```yaml
+head:
+  semconv_version: n
+baseline:
+  semconv_version: n+1
+changes:
+  attributes:
+    - name: process.cpu.state
+      type: renamed
+      new_name: cpu.mode
+      note: Deprecated, use `cpu.mode` instead.    
+```
+
+### Conditional renaming
+
+Sometimes, the renaming of an item is context-dependent. Let's consider the following
+two consecutive versions of a semantic convention registry:
+
+```yaml
+# Version n
+groups:
+  - id: registry.network
+    type: attribute_group
+    attributes:
+      - id: net.peer.name
+        type: string
+        stability: experimental
+        examples: ['example.com']
+        #...
+```
+
+To rename the `net.peer.name` attribute to `server.address` on client spans and
+`client.address` on server spans, the following changes are made:
+
+```yaml
+# Version n+1
+groups:
+  - id: registry.network.deprecated
+    type: attribute_group
+    attributes:
+      - id: net.peer.name
+        type: string
+        brief: Deprecated, use `server.address` on client spans and `client.address` on server spans.
+        deprecated:
+          type: conditionally_renamed
+          forward: >
+            switch span_kind {
+              case 'client' => attributes['server.address'] = attributes['net.peer.name'],
+              case 'server' => attributes['client.address'] = attributes['net.peer.name']
+            }
+          backward: >
+            switch span_kind {
+              case 'client' => attributes['net.peer.name'] = attributes['server.address'],
+              case 'server' => attributes['net.peer.name'] = attributes['client.address']
+            }
+        stability: experimental
+        examples: ['example.com']
+```
```

Comment:
In this example, the attribute has been deprecated in the attribute_group but the condition is related to its usage in a span. Should the forward / backward instructions be on the span definition where `net.peer.name` is referenced? There may be different instructions for different spans that use `net.peer.name`.

Reply:
Good point. It's a good topic for discussion for our next Semantic Conventions Tooling SIG.

Reply:
Following the discussion in the last SIG meeting, it was decided to start with a more minimalist approach initially. As a result, I have updated this PR accordingly, both in the code and the main documentation (not this doc).

However, I am keeping this exploration document in place so that we can revisit this topic later if needed.